packages/common-ui/lib/list-page/QueryRow.tsx

```diff
@@ -28,49 +25,113 @@ export interface QueryRowProps {
+   * their own field where it's stored.
+   */
   matchValue?: string;
-  fieldRangeStart?: string;
-  fieldRangeEnd?: string;
-  matchType?: string;
-  numericalMatchType?: string;
-  compoundQueryType?: string;
+
+  /**
+   * Changes based on the type of the field.
+   *
+   * Possible Options: "equals", "contains", "notEquals", "empty", "notEmpty", "greaterThan",
+   *                   "lessThan", "greaterThanOrEqualTo", "lessThanOrEqualTo"
+   */
+  matchType?:
+    | "equals"
+    | "contains"
+    | "notEquals"
+    | "empty"
+    | "notEmpty"
+    | "greaterThan"
+    | "lessThan"
+    | "greaterThanOrEqualTo"
+    | "lessThanOrEqualTo";
+
+  /**
+   * For text based searches you can change the match type to show only when the search is exactly
+   * the same as the matchValue or partial match.
+   *
+   * Possible Options: "exact" and "partial".
+   */
+  textMatchType?: "exact" | "partial";
+
+  /**
+   * Currently, only "and" is supported. In the future, "or" can be selected.
+   *
+   * Possible Options: "and" and "or".
+   */
+  compoundQueryType?: "and" | "or";
+
+  /**
+   * When searching numbers, the value will be stored here. Only if the type is "number".
+   */
   number?: string;
+
+  /**
+   * When searching dates, the value will be stored here. Only if the type is "date".
+   */
   date?: string;
+
+  /**
+   * When searching booleans, the value will be stored here. Only if the type is "boolean".
+   */
   boolean?: string;
+
+  /**
+   * Based on the field selected, the type of that field is stored here.
+   */
   type?: string;
+
+  /**
+   * If the field is a relationship, this is the relationship name.
+   *
+   * Example: "author"
+   */
   parentName?: string;
+
+  /**
+   * If the field is a relationship, this is the relationship type.
+   *
+   * Example: "agent"
+   */
   parentType?: string;
+
+  /**
+   * If the field is a relationship, this is the relationship field being used.
+   *
+   * Example: "firstName"
+   */
   parentPath?: string;
+
+  /**
+   * Is the value being searched on a distinct field, if so, suggestions can be provided to help
+   * the user find what they are searching for.
+   */
   distinctTerm?: boolean;
 }
 
-const queryRowMatchOptions = [
-  { label: "PARTIAL_MATCH", value: "match" },
-  { label: "EXACT_MATCH", value: "term" }
-];
-
-const queryRowNumericalMatchOptions = (isDateField: boolean) => {
-  const options = [
-    { label: "Equal to", value: "equal" },
-    { label: "Greater than", value: "greaterThan" },
-    { label: "Greater than or equal to", value: "greaterThanEqual" },
-    { label: "Less than", value: "lessThan" },
-    { label: "Less than or equal to", value: "lessThanEqual" }
-  ];
-
-  // Only the data field should contain this.
-  if (isDateField) {
-    options.splice(1, 0, { label: "Contains", value: "contains" });
-  }
-
-  return options;
-};
-
-const queryRowBooleanOptions = [
-  { label: "TRUE", value: "true" },
-  { label: "FALSE", value: "false" }
-];
```

Comment:
Instead of hosting all the different types on the `QueryRow.tsx` file, they have been moved to their own individual files.

Checkout the query-row-search-options folder with the 5 different types. Such as boolean and date types.

As more types get introduced for the search it can be added to this folder. For example, location searching.